src/workerd/api/node/zlib-util.h

```diff
@@ -250,38 +286,111 @@ class ZlibUtil final: public jsg::Object {
         kj::Array<kj::byte> output,
         int outputOffset,
         int outputLength);
-    void writeSync(jsg::Lock& js,
-        int flush,
-        jsg::Optional<kj::Array<kj::byte>> input,
-        int inputOffset,
-        int inputLength,
-        kj::Array<kj::byte> output,
-        int outputOffset,
-        int outputLength);
-    void params(jsg::Lock& js, int level, int strategy);
     void reset(jsg::Lock& js);
 
+    JSG_RESOURCE_TYPE(CompressionStream) {
+      JSG_METHOD(close);
+      JSG_METHOD_NAMED(write, template write<true>);
+      JSG_METHOD_NAMED(writeSync, template write<false>);
+      JSG_METHOD(reset);
+      JSG_METHOD(setErrorHandler);
+    }
+
+  protected:
+    CompressionContext* context() {
```

Comment:
Is this a pointer because node? I'd rather use a reference

Reply:
This follows Node.js implementation but this is actually needed in `BrotliCompressionStream<CompressionContext>::initialize` function.

The use case is `this->context()->initialize` where if we define this context as a protected variable and expose it, the compiler will complain, since both BrotliCompressionStream and CompressionStream has context variables. (PS: BrotliCompressionStream extends CompressionStream)

Reply:
I don't quite get why there would be two of them. I see only one variable named context_

Reply:
I'll open a PR after this pull-request lands, to show the build error. For now, if this is not blocking, let's proceed with the PR. wdyt @npaun?

Reply:
Not blocking. I'll start tagging my comments with _nit:_ for clarity.